nextflow_schema.json

```diff
@@ -13,6 +13,10 @@
             "properties": {
                 "input": {
                     "type": "string",
+                    "format": "file-path",
```

Comment:
Maybe we need to have another folder for all of these JSON schema files now that we have more than one?

Reply:
Long term this is a good idea I think, but will need a larger coordinated effort as currently all tooling expects a root-level `nextflow_schema.json` file. So if we move that then tonnes of stuff will break.

Reply:
Now is a better time than any? Could ignore the lint for it and fix in the next release given the Schema stuff is still bedding in? May also be nice to rename to `schema_nextflow.json` for consistency with the other ones? Or maybe we keep the main Schema in the top-level directory and just add the samplesheet JSON files to `schema/` ? Then we can can just have them called by parameter name e.g. `input.json` and `public_data_ids.json` ? Not 💯 with these names either to be honest. Any thoughts?

Reply:
Mainly because they may not be descriptive enough with just the parameter name as to what the Schemas contain. e.g. is something like `samplesheet_input.json` and `samplesheet_public_data_ids.json` better?

Reply:
I saw used both `file-path` and `path`, what's the difference ?

Reply:
@drpatelh - sure, but the `nf-core schema build` and `nf-core launch` cli functionality will break. Also the website _Launch_ functionality and the website parameter documentation. Also the pipeline parameter validation. So it's not just one or two linting tests 😉 

Regarding the sample sheet schema files, I have structured this PR so that files can be called whatever you want - the `schema` keywords is a file path that is relative to the main schema. I did wonder about just defaulting to `schema_<parameter-name>.json` but figured that there would likely be use cases where there could be multiple different params that use similar file types and want to share the same schema.

Given the above, I think that my preference would be to keep `nextflow_schema.json` where it is, as we already have so much built around that. Where and what we call the other files I don't really mind, as the file paths don't (indeed can't) be known in advance and are always discovered from the main `nextflow_schema.json` file. So there will never be loads of infrastructure built around their filenames etc and we can be as flexible as we like with those. Does that make sense?

---

@pditommaso - `path` can be a file or a directory, `file-path` and `directory-path` are specific to one or the other. We can call these whatever we like, but I have taken them from the `pydantic` library as I thought that they were quite good and like to stick to pre-existing standards if I can.

See https://pydantic-docs.helpmanual.io/usage/schema/#json-schema-types

Reply:
> So there will never be loads of infrastructure built around their filenames etc and we can be as flexible as we like with those. Does that make sense?

It does indeed. No harm in thinking about this from the offset though!